components/nimbus/tests/test_restart.rs

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+mod common;
+#[cfg(feature = "rkv-safe-mode")]
+#[cfg(test)]
+mod test {
+
+    use super::common::new_test_client_with_db;
+    #[cfg(feature = "rkv-safe-mode")]
+    use nimbus::error::Result;
+    use tempdir::TempDir;
+
+    #[test]
+    fn test_restart_opt_in() -> Result<()> {
```

Comment:
This tests the restart scenario as close as possible - I'm still a bit fuzzy about the details of `apply_pending_experiments`, but I think I get the big idea. 

This also passes without the change because of the same reason.

Reply:
> This also passes without the change because of the same reason.

🧙 If you can't reproduce the problem, how do you know you've fixed it? 🧙‍♂️